lib/config/optionsconfiguration.go

```diff
@@ -6,43 +6,132 @@
 
 package config
 
+import (
+	"encoding/json"
+	"encoding/xml"
+	"fmt"
+)
+
+type WeakHashSelectionMethod int
+
+const (
+	WeakHashAuto WeakHashSelectionMethod = 1 << iota
```

Comment:
No need for bitmasking here I think; just `= iota` and the default value becomes zero, no SetDefaults shenanigans required.